### PR TITLE
ipv6_nc: add iterators

### PR DESCRIPTION
**Pull request by authmillenon**
Adds iterators for the neighbor cache. Inspired by iterators in `utlist`.

### CODE REVIEW

File: sys/net/network_layer/ng_ipv6/nc/ng_ipv6_nc.c [hunk 1]
```diff
@@ -136,6 +136,50 @@ ng_ipv6_nc_t *ng_ipv6_nc_get(kernel_pid_t iface, const ng_ipv6_addr_t *ipv6_addr
...
+ng_ipv6_nc_t *ng_ipv6_nc_get_next(ng_ipv6_nc_t *prev)
+{
+    if (prev == NULL) {
+        prev = ncache;
+    }
+    else {
+        prev++;     /* get next entry */
+    }
+
+    while (prev - ncache < NG_IPV6_NC_SIZE) {   /* while not reached end */
```
Comment (cgundogan) [inline sys/net/network_layer/ng_ipv6/nc/ng_ipv6_nc.c]: changing the inequality to `prev < NG_IPV6_NC_SIZE + ncache` might be preferable. In this case the cpu might cache the right hand in one of its registers, hence computing it only once.

Comment (authmillenon) [inline sys/net/network_layer/ng_ipv6/nc/ng_ipv6_nc.c]: Done

**Comment by haukepetersen** [comment]
I don't like the use of the ITER... macros here. Except having to write 3 lines less of code for iterating (and no saving of code size whatsoever), these macros don't have any other benefit. So far we were able to keep RIOT largely free of function-style-macros, and I think we should keep it that way!

**Comment by authmillenon** [comment]
This is an iterator macro, not a function-style-macro...

**Comment by authmillenon** [comment]
Okay. I'll remove the macros.

**Comment by authmillenon** [comment]
Done

**Comment by authmillenon** [comment]
Rebased to current master and adapted shell commands for the iterators (adds commands for listing neighbors and routers)

**Comment by OlegHahm** [comment]
I think this PR does a little bit more than its description indicates.

**Comment by authmillenon** [comment]
I don't think so, but please tell me what is too much.

**Comment by OlegHahm** [comment]
I was referring mostly to the shell command.

**Comment by authmillenon** [comment]
That's just extends the shell command by this particular functionality in this PR. I wanted to add this in #2583 already, but I wanted to keep the dependencies of that one low, so I retrofitted it in here. In light of #2755 it might be a good idea however to make `ncache [list]` and `ncache add|del` two separate commands ;-)

**Comment by authmillenon** [comment]
Ping?

**Comment by OlegHahm** [comment]
ACK

**Comment by authmillenon** [comment]
Squashed

**Pull request merged**